File: Data Structures & Algorithms/two-integer-sum/submission-1.go
func twoSum(nums []int, target int) []int {
	m := map[int]int{}
	for i, n := range nums {
		diff := target - n
		if j, ok := m[diff]; ok {
			if i > j {
				return []int{j, i}
			}
			if j > i {
				return []int{i, j}
			}
		}
		m[n] = i
	}

	return []int{}
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(arr, target) {
        let left = 0, right = arr.length - 1;
        while(left < right){
            const sum = arr[left] + arr[right];
            if(sum == target) return [left + 1, right + 1];
            if(sum > target) right--;
            else left++;
        }
        return [];
    }
}
